pkg/controller/nodenetworkstatereport/nodenetworkstatereport_controller.go

```diff
@@ -104,25 +105,28 @@ type ReconcileNodeNetworkStateReport struct {
 func (r *ReconcileNodeNetworkStateReport) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.V(1).Info("Reconciling NodeNetworkState report")
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
This retries only when we have API write conflict, but returns IsNotFound error as soon as it happens? Neat.